components/airtable/new-modified-or-deleted-records.js

```diff
@@ -0,0 +1,111 @@
+const airtable = require('https://github.com/PipedreamHQ/pipedream/components/airtable/airtable.app.js');
+const moment = require('moment');
+const axios = require('axios');
+const Bottleneck = require('bottleneck');
+
+const limiter = new Bottleneck({
+  minTime: 200, // 5 requets per second
+});
+const axiosRateLimiter = limiter.wrap(axios);
+
+module.exports = {
+  name: 'New, Modified or Deleted Records',
+  version: '0.0.2',
+  props: {
+    db: '$.service.db',
+    airtable,
+    baseId: { type: '$.airtable.baseId', appProp: 'airtable' },
+    tableId: { type: '$.airtable.tableId', baseIdProp: 'baseId' },
+    timer: {
+      type: '$.interface.timer',
+      default: {
+        intervalSeconds: 60 * 5,
+      },
+    },
+  },
+  async run(event) {
+    const { baseId, tableId, viewId } = this;
+    const metadata = {
+      baseId,
+      tableId,
+      viewId,
+    };
+
+    const config = {
+      url: `https://api.airtable.com/v0/${encodeURIComponent(this.baseId)}/${encodeURIComponent(this.tableId)}`,
+      params: {},
+      headers: {
+        Authorization: `Bearer ${this.airtable.$auth.api_key}`,
+      },
+    };
+
+    const lastTimestamp = this.db.get('lastTimestamp');
+    const prevAllRecordIds = this.db.get('prevAllRecordIds');
```

Comment:
We currently don't impose limits on the size of `this.db`, but we may in the future. I could see this list growing large for some tables. Chatting with the team about this.

Reply:
Since we are only saving the ids, would this amount to a significant amount of disk space.

Reply:
Yes, each record ID in Airtable is a string of ~17 bytes, so if a user had a base / table with 50,000 records, for example (limit on the number of records of the Pro plan), that's 850Kb of data, which may end up being higher than the limit we enforce for the total size of `this.db` for a given component.

But for now, this is fine - I have another suggested change to the `description` that should warn users with large bases.